### PR TITLE
complete task 6

### PR DESCRIPTION
**Pull request by winniekwoknorth**

### CODE REVIEW

File: __tests__/tests.test.js [hunk 1]
```diff
@@ -159,3 +159,49 @@ describe("/api/articles/", () => {
...
+        expect(res.body.comments.length).toBe(2);
+        expect(res.body.comments).toBeSortedBy("created_at", { descending: true });
+        expect(res.body.comments[0]).toMatchObject( {
+          comment_id: 11,
+          body: 'Ambidextrous marsupial',
+          article_id: 3,
+          author: 'icellusedkars',
+          votes: 0,
+          created_at: '2020-09-19T23:10:00.000Z'
+        })
```
Comment (MKD01) [inline __tests__/tests.test.js]: We dont need to check the first comment specifically here as you are checking every comment below this, so this expect isnt needed.

File: __tests__/tests.test.js [hunk 2]
```diff
@@ -159,3 +159,49 @@ describe("/api/articles/", () => {
...
+  })
+  test.only("get 404, not found if request is not found in database", () => {
+    return request(app)
+      .get("/api/articles/999/comments")
+      .expect(404)
+      .then((res) => {
+        expect(res.body.msg).toBe("comments does not exist");
+      });
+  })
+})
```
Comment (MKD01) [inline __tests__/tests.test.js:208]: Really good testing so far, but one final test that should also be added here is, when a user sends a request to a valid article id but one that does not have any comments, what should we respond with for this?

**Review by MKD01 — CHANGES_REQUESTED** [review]
Good work so far, but I would like to see that final test before you merge this in, so please re-submit this pr once that test has been added.

File: controllers/articles.controllers.js [hunk 3]
```diff
@@ -16,3 +16,10 @@ exports.getArticles = (req, res, next) => {
        .catch(next);
 };
 
+// exports.getCommentsByArticleId = (req, res, next) => {
+//     const { article_id } = req.params
+//     selectComments(article_id).then((comments) => {
+//      res.status(200).send({ comments });
+//     })
+//        .catch(next);
+// }
```
Comment (MKD01) [inline controllers/articles.controllers.js]: Dont forget to remove commented out code.

File: __tests__/tests.test.js [hunk 4]
```diff
@@ -159,3 +159,50 @@ describe("/api/articles/", () => {
...
+      });
+  })
+  test("get 200, and return empty array if no comment for a valid request of article_id", () => {
+    return request(app)
+      .get("/api/articles/2/comments")
+      .expect(200)
+      .then((res) => {
+        expect(typeof res.body).toBe("object");
+        
+      });
```
Comment (MKD01) [inline __tests__/tests.test.js]: This test description is really good, but you are not actually checking that the array is empty here.

**Review by MKD01 — APPROVED** [review]
Really good work, but you are missing an expect in the new test you have added. Once that has been added I am happy for you to merge this in.

File: __tests__/tests.test.js (hunk 1 above)
Comment (winniekwoknorth) [reply __tests__/tests.test.js]: deleted this part

File: __tests__/tests.test.js (hunk 4 above)
Comment (winniekwoknorth) [reply __tests__/tests.test.js]: updated to check for []

**Pull request merged**